src/main/java/com/iridium/iridiumskyblock/BlockData.java

```diff
@@ -50,15 +50,20 @@ public void setBlock(Block block) {
         BlockState blockState = block.getState();
         Material mat = material.parseMaterial();
         if (mat != null) {
-            blockState.setType(mat);
-            blockState.setRawData(data);
-            blockState.update(true, true);
+            settingBlock(block, blockState, mat);
```

Comment:
I dont see why your creating a method for this, especially since its only called in one place?